README.md

```diff
@@ -1,93 +1,134 @@
+Other than being fast, this method has an additional perk - if you need to remove elements as you iterate, looping backwards may eliminate the need to implement additional low-level logic to accommodate item removal.
 
-```js
-for(;;);
-```
+It's also possible to move all the code between the braces into the final `iteration` expression of the `for` loop if need be. However, note that common keywords and tokens such as `if` and `break` cannot be used in the middle of expressions. For complex logic, use [ternaries and logical operators](#condensing-conditionals) instead.
 
-## Empty expressions
+After moving all the code out of a loop block, you may use a semicolon to convert the resulting empty block into an empty expression:
 
-Instead of empty blocks:
+```javascript
+// Empty block `{}`
+for(;;){}
 
-```js
-while (condition) {}
+// ...becomes an "empty expression":
+for(;;);
 ```
 
-Create "empty expressions":
+The semicolon is required at the end of the parentheses if it is followed by an expression that you wish to keep "out of the loop".
+
+```javascript
+// `return` statement part of `for` loop, will break out of function on first iteration
+for(var r='';...;...)return r
 
-```js
-while(condition);
+// `return` occurs after the `for` loop (desired result)
+for(var r ='';...;...);return r
 ```
 
-## Blocks to sequence expressions
+**Note:** Even if an expression in a `for` loop is empty, both semicolons are still required or else the compiler will throw a `SyntaxError`. This is one of the reasons why these kinds of loops struggle with readability.
 
-When normally writing code, you'd do something like this:
+**Another one:** `for(;;)` is the smallest infinite loop possible.
 
-```js
-if (condition) {
-  foo()
-  bar()
-  baz()
-}
-```
+## Chaining variable declarations
+Defining variables in separate declarations can help with readability, but the definitions can also be placed onto one line to save space.
 
-These can be changed to a "sequence expression":
+```javascript
+// Sequence of variable declarations
+var i = 0
+var lorem = 'ipsum'
+var foo
 
-```js
-if(condition)foo(),bar(),baz() 
+// ...becomes a "declaration sequence":
+var i=0,l='ipsum',f
 ```
 
-## If statements to conditional expressions
+Incidentally, this method fits in naturally into the `initial` expression of `for` loops since keywords like `var` cannot be used in sequence expressions.
 
-An `if` (with no `else`):
+## Condensing conditionals
+Most `if` statements can be condensed into plain expressions by using ternaries or logical operators, allowing you to use them in `for` loops.
 
-```js
+```javascript
+// Generic `if` statement
 if (condition) {
   consequent()
 }
-```
 
-Can be changed to:
-
-```js
+// ...becomes a "logical expression":
 condition&&consequent()
 ```
 
-And one with `else`:
-
-```js
+```javascript
+// `if`-`else` statement
 if (condition) {
   consequent()
 } else {
   alternate()
 }
+
+// ...becomes a "ternary expression":
```

Comment:
Being picky here, but since this repo is only a README I'll be strict on the wording.

These are called ["conditional expressions"](https://github.com/estree/estree/blob/master/es5.md#conditionalexpression).  Which makes more sense to me.  Also that "ternary expression" has redundancy to it.